crates/rune/src/tests/deprecation.rs

```diff
@@ -61,17 +61,15 @@ fn check_diagnostic(context: &Context, diagnostic: &Diagnostic, msg: &str) {
             WarningDiagnosticKind::UsedDeprecated { message, .. } => {
                 assert_eq!(message, msg);
             }
-            _ => panic!("Expected a UsedDeprecated warning"),
+            kind => panic!("Unexpected warning: {kind:?}"),
         },
         Diagnostic::RuntimeWarning(w) => match w.kind() {
             RuntimeWarningDiagnosticKind::UsedDeprecated { hash, .. } => {
                 let message = context.lookup_deprecation(*hash);
                 assert_eq!(message, Some(msg));
             }
-            #[allow(unreachable_patterns)]
-            _ => panic!("Expected a UsedDeprecated warning"),
         },
-        _ => panic!("Diagnostic need to be of typ warning"),
+        kind => panic!("Unexpected diagnostics: {kind:?}"),
```

Comment:
In case this assertion fails, we'll probably need to know how it fails to fix the error.

There is a better method available called assert_errors! which does a lot of this by default, but it doesn't cleanly plug into how you're testing the context.